plugins/inputs/prometheus/parser.go

```diff
@@ -21,6 +21,145 @@ import (
+				// summary metric
+				telegrafMetrics := makeQuantilesV2(m, tags, metricName, mf.GetType())
+				metrics = append(metrics, telegrafMetrics...)
+			} else if mf.GetType() == dto.MetricType_HISTOGRAM {
+				// histogram metric
+				telegrafMetrics := makeBucketsV2(m, tags, metricName, mf.GetType())
+				metrics = append(metrics, telegrafMetrics...)
+			} else {
+				// standard metric
+				// reading fields
+				fields := make(map[string]interface{})
+				fields = getNameAndValueV2(m, metricName)
+				// converting to telegraf metric
+				if len(fields) > 0 {
+					var t time.Time
+					if m.TimestampMs != nil && *m.TimestampMs > 0 {
+						t = time.Unix(0, *m.TimestampMs*1000000)
+					} else {
+						t = time.Now()
+					}
+					metric, err := metric.New("prometheus", tags, fields, t, valueType(mf.GetType()))
+					if err == nil {
+						metrics = append(metrics, metric)
+					}
+				}
+			}
+		}
+	}
+
+	return metrics, err
+}
+
+// Get Quantiles for summary metric & Buckets for histogram
+func makeQuantilesV2(m *dto.Metric, tags map[string]string, metricName string, metricType dto.MetricType) []telegraf.Metric {
+	var metrics []telegraf.Metric
+	fields := make(map[string]interface{})
+	var t time.Time
+	if m.TimestampMs != nil && *m.TimestampMs > 0 {
+		t = time.Unix(0, *m.TimestampMs*1000000)
+	} else {
+		t = time.Now()
+	}
+	fields[metricName+"_count"] = float64(m.GetSummary().GetSampleCount())
+	fields[metricName+"_sum"] = float64(m.GetSummary().GetSampleSum())
+	met, err := metric.New("prometheus", tags, fields, t, valueType(metricType))
+	if err == nil {
+		metrics = append(metrics, met)
+	}
+
+	for _, q := range m.GetSummary().Quantile {
+		newTags := tags
+		fields = make(map[string]interface{})
+		if !math.IsNaN(q.GetValue()) {
+			newTags["quantile"] = fmt.Sprint(q.GetQuantile())
+			fields[metricName] = float64(q.GetValue())
+
+			quantileMetric, err := metric.New("prometheus", newTags, fields, t, valueType(metricType))
+			if err == nil {
+				metrics = append(metrics, quantileMetric)
+			}
+		}
+	}
+	return metrics
+}
+
+// Get Buckets  from histogram metric
+func makeBucketsV2(m *dto.Metric, tags map[string]string, metricName string, metricType dto.MetricType) []telegraf.Metric {
+	var metrics []telegraf.Metric
+	fields := make(map[string]interface{})
+	var t time.Time
+	if m.TimestampMs != nil && *m.TimestampMs > 0 {
+		t = time.Unix(0, *m.TimestampMs*1000000)
+	} else {
+		t = time.Now()
+	}
+	fields[metricName+"_count"] = float64(m.GetHistogram().GetSampleCount())
+	fields[metricName+"_sum"] = float64(m.GetHistogram().GetSampleSum())
+
+	met, err := metric.New("prometheus", tags, fields, t, valueType(metricType))
+	if err == nil {
+		metrics = append(metrics, met)
+	}
+
+	for _, b := range m.GetHistogram().Bucket {
+		newTags := tags
+		fields = make(map[string]interface{})
+		newTags["le"] = fmt.Sprint(b.GetUpperBound())
+		fields[metricName] = float64(b.GetCumulativeCount())
```

Comment:
I think we need to append `_bucket` to the metricName, to preserve the original name.  From my testing this suffix is removed...

Reply:
@danielnelson - are you saying every histogram metric will have _bucket appended to it as part of this parsing ?

Reply:
Excluding the `_sum` and `_count`.   In my testing this seems to be required in order to keep the metric name as is, otherwise the `_bucket` suffix is lost.

Reply:
OK thanks. I will look into it tomorrow and fix and test it.

Reply:
@danielnelson - fixed.
Now we are collecting all (sum,count & bucket) as is . Below is what i checked for a k8s api server histogram -

apiserver_request_latencies_bucket{resource="pods",subresource="",verb="LIST",le="+Inf"} 2
apiserver_request_latencies_sum{resource="pods",subresource="",verb="LIST"} 1682
apiserver_request_latencies_count{resource="pods",subresource="",verb="LIST"} 2